hedera-node/src/main/java/com/hedera/services/txns/schedule/ScheduleProcessing.java

```diff
@@ -0,0 +1,288 @@
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.hedera.services.context.properties.GlobalDynamicProperties;
+import com.hedera.services.ledger.SigImpactHistorian;
+import com.hedera.services.state.merkle.MerkleScheduledTransactions;
+import com.hedera.services.state.submerkle.RichInstant;
+import com.hedera.services.state.virtual.schedule.ScheduleVirtualValue;
+import com.hedera.services.store.schedule.ScheduleStore;
+import com.hedera.services.throttling.TimedFunctionalityThrottling;
+import com.hedera.services.throttling.annotations.ScheduleThrottle;
+import com.hedera.services.utils.EntityNum;
+import com.hedera.services.utils.accessors.SignedTxnAccessor;
+import com.hedera.services.utils.accessors.TxnAccessor;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.ScheduleID;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.eclipse.collections.impl.set.mutable.primitive.LongHashSet;
+
+import static com.hedera.services.utils.EntityNum.fromScheduleId;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SCHEDULE_FUTURE_GAS_LIMIT_EXCEEDED;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SCHEDULE_FUTURE_THROTTLE_EXCEEDED;
+
+/**
+ * Class that encapsulates some of the more complex processing of scheduled transactions.
+ */
+@Singleton
+public class ScheduleProcessing {
+	private static final Logger log = LogManager.getLogger(ScheduleProcessing.class);
+
+	private final SigImpactHistorian sigImpactHistorian;
+	private final ScheduleStore store;
+	private final ScheduleExecutor scheduleExecutor;
+	private final GlobalDynamicProperties dynamicProperties;
+	private final TimedFunctionalityThrottling scheduleThrottling;
+	private final Supplier<MerkleScheduledTransactions> schedules;
+	private final long maxProcessingLoopIterations;
+
+	SigMapScheduleClassifier classifier = new SigMapScheduleClassifier();
+	SignatoryUtils.ScheduledSigningsWitness signingsWitness = SignatoryUtils::witnessScoped;
+	Predicate<ScheduleVirtualValue> isFullySigned;
+
+	@Inject
+	public ScheduleProcessing(final SigImpactHistorian sigImpactHistorian, final ScheduleStore store,
+			final ScheduleExecutor scheduleExecutor, final GlobalDynamicProperties dynamicProperties,
+			final ScheduleSigsVerifier scheduleSigsVerifier,
+			@ScheduleThrottle final TimedFunctionalityThrottling scheduleThrottling,
+			final Supplier<MerkleScheduledTransactions> schedules) {
+		this.sigImpactHistorian = sigImpactHistorian;
+		this.store = store;
+		this.scheduleExecutor = scheduleExecutor;
+		this.dynamicProperties = dynamicProperties;
+		this.scheduleThrottling = scheduleThrottling;
+		isFullySigned = scheduleSigsVerifier::areAllKeysActive;
+		this.schedules = schedules;
+		this.maxProcessingLoopIterations = dynamicProperties.schedulingMaxTxnPerSecond() * 2;
```

Comment:
Seems unlikely this setting will need to be updated dynamically, but setting the field once here eliminates our option.